pkg/snapshotUtils/snapshot.go

```diff
@@ -48,39 +50,31 @@ func checkPhasesAndSendResult(waitForPhases []backupdriverv1.SnapshotPhase, snap
 	}
 }
 
-func SnapshopRef(ctx context.Context, clientSet *v1.BackupdriverV1Client, objectToSnapshot core_v1.TypedLocalObjectReference, namespace string, repository BackupRepository, waitForPhases []backupdriverv1.SnapshotPhase, logger logrus.FieldLogger) (backupdriverv1.Snapshot, error) {
-	snapshotUUID, err := uuid.NewRandom()
+// Create a Snapshot record in the specified namespace.
+func SnapshotRef(ctx context.Context,
+	clientSet *v1.BackupdriverV1Client,
+	objectToSnapshot core_v1.TypedLocalObjectReference,
+	namespace string,
+	repository BackupRepository,
+	waitForPhases []backupdriverv1.SnapshotPhase,
+	logger logrus.FieldLogger) (backupdriverv1.Snapshot, error) {
 
+	snapshotUUID, err := uuid.NewRandom()
```

Comment:
So snapshotUUID is still generated inside SnapshotRef.  How do you make it is not generated again for the same request?

Reply:
That was fixed by Lintong's commit - https://github.com/vmware-tanzu/velero-plugin-for-vsphere/commit/865897be99637a8626714346c922df5f0fe0b174